mesonbuild/backend/backends.py

```diff
@@ -756,17 +770,34 @@ def rpaths_for_non_system_absolute_shared_libraries(self, target: build.BuildTar
                 if os.path.splitext(libpath)[1] not in ['.dll', '.lib', '.so', '.dylib']:
                     continue
 
-                try:
-                    commonpath = os.path.commonpath((libdir, srcdir))
-                except ValueError: # when paths are on different drives on Windows
-                    commonpath = ''
+                if mesonlib.is_windows() and libpath.endswith('.lib'):
+
+                    if isinstance(dep, dependencies.PkgConfigDependency):
+                        # If by chance pkg-config knows the bin dir...
+                        # Hoping conan will add it to its generated pc files.
+                        found = False
+                        for index in count(1):
+                            bindir = dep.get_pkgconfig_variable(f'bindir{index}', [], default='')
+                            if bindir:
+                                found = True
+                                add_path(bindir)
+                                continue
+                            break
+                        if found:
+                            continue
+
+                    # If the dll is not in the same dir, try to replace 'lib' with 'bin' in the path.
+                    # This is not 100% proof, since the bin dir could be elsewhere or with a different
+                    # name, and the DLL could have a different name as well, but this is better than nothing.
+                    p = Path(libpath)
+                    if not p.with_suffix('.dll').exists():
+                        # replace last occurence:
+                        binpath = '/bin/'.join(p.as_posix().rsplit('/lib/', maxsplit=1))
```

Comment:
Isn't this equivalent to more readable `p.as_posix().replace('/lib/', '/bin/', maxreplace=1)`?

Reply:
No... It is equivalent to `rreplace`, but `rreplace` is not a Python function... It replaces the **last** occurrence in the string. Just in case the is another `lib` subdirectory in the path.

Reply:
well, if there are more than one occurrence it's arbitrary which is needs to be replaced. Why would last be better than first? But ok, your way works too.

Reply:
Well. Maybe I'm overthinking and replace all would be just as good...

The idea is that the package is usually installed as
```
/path/to/mylib/lib/mylib.lib
/path/to/mylib/bin/mylib.dll
```
or
```
/path/to/mylib/lib/x64/mylib.lib
/path/to/mylib/bin/x64/mylib.dll
```

so the last occurence seems to be the safest bet.

Reply:
Luckily `/mylib/` won't match `/lib/`. But as you prefer, it was just a small nitpicking.

Reply:
In case you still weren't sure I can resolve this once and for all. :)

```python
>>> p = Path('foo/bar')
>>> p.as_posix().replace('/lib/', '/bin/', maxreplace=1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: str.replace() takes no keyword arguments
```